plansys2_problem_expert/src/plansys2_problem_expert/ProblemExpert.cpp

```diff
@@ -132,6 +132,8 @@ ProblemExpert::removePredicate(const Predicate & predicate)
   bool found = false;
   int i = 0;
 
+  if (!isValidPredicate(predicate)) // if predicate is not valid, error
+    return false;
```

Comment:
Same behavior should be added to removeAssignment (for behavior homogeneity)

Reply:
@teyssieuman not sure that I understand the use of add/remove assignment. removeAssignment is not used anywhere in the code base, not even the terminal application. What is the purpose of assignments?

Reply:
An assignment is a function assignment:
(= (Function_name parameter1 ... parameterN) value)

It can be added, removed or changed (giving a new value)...

I submitted a pull request to your branche.

Reply:
Thanks. This is my first time actually merging code with a public repo. So patience on me getting up-to-speed is appreciated. Note that as pointed out by others, my break statements did not actually speed anything up and I removed them. Fixed testing and pushed a new commit now that includes the merge